src/Business Foundation/App/NoSeries/src/Legacy/NoSeriesManagement.Codeunit.al

```diff
@@ -0,0 +1,968 @@
+    procedure TestManualWithDocumentNo(DefaultNoSeriesCode: Code[20]; DocumentNo: Code[20])
+    begin
+        if DefaultNoSeriesCode <> '' then begin
+            GlobalNoSeries.Get(DefaultNoSeriesCode);
+            if not GlobalNoSeries."Manual Nos." then
+                Error(PostErr, DocumentNo);
+        end;
+    end;
+
+    [Obsolete('This function is used for compatibility with extension usages of the old OnBeforeInitSeries event. Now the new No. Series is used. InitSeries no longer exist, instead a No. Series is selected and the next number is retrieved.', '24.0')]
+    procedure RaiseObsoleteOnBeforeInitSeries(var DefaultNoSeriesCode: Code[20]; OldNoSeriesCode: Code[20]; NewDate: Date; var NewNo: Code[20]; var NewNoSeriesCode: Code[20]; var IsHandled: Boolean)
+    begin
+        OnBeforeInitSeries(DefaultNoSeriesCode, OldNoSeriesCode, NewDate, NewNo, NewNoSeriesCode, GlobalNoSeries, IsHandled, GlobalNoSeriesCode);
+    end;
+
+    [Obsolete('This function is used for compatibility with extension usages of the old OnAfterInitSeries event. Now the new No. Series is used. InitSeries no longer exist, instead a No. Series is selected and the next number is retrieved.', '24.0')]
+    procedure RaiseObsoleteOnAfterInitSeries(NoSeriesCode: Code[20]; DefaultNoSeriesCode: Code[20]; NewDate: Date; var NewNo: Code[20])
+    var
+        NoSeries: Record "No. Series";
+    begin
+        if NoSeries.Get(NoSeriesCode) then;
+        OnAfterInitSeries(NoSeries, DefaultNoSeriesCode, NewDate, NewNo);
+    end;
+
+    [Obsolete('Please use AreRelated in the "No. Series" codeunit instead', '24.0')]
+    procedure InitSeries(DefaultNoSeriesCode: Code[20]; OldNoSeriesCode: Code[20]; NewDate: Date; var NewNo: Code[20]; var NewNoSeriesCode: Code[20])
+    var
+        IsHandled: Boolean;
+    begin
+        IsHandled := false;
+        OnBeforeInitSeries(DefaultNoSeriesCode, OldNoSeriesCode, NewDate, NewNo, NewNoSeriesCode, GlobalNoSeries, IsHandled, GlobalNoSeriesCode);
+        if IsHandled then
+            exit;
+
+        if NewNo = '' then begin
+            GlobalNoSeries.Get(DefaultNoSeriesCode);
+            if not GlobalNoSeries."Default Nos." then
+                Error(CannotAssignAutomaticallyErr, GlobalNoSeries.FieldCaption("Default Nos."), GlobalNoSeries.TableCaption(), GlobalNoSeries.Code);
+            if OldNoSeriesCode <> '' then begin
+                GlobalNoSeriesCode := DefaultNoSeriesCode;
+                FilterSeries();
+                GlobalNoSeries.Code := OldNoSeriesCode;
+                if not GlobalNoSeries.Find() then
+                    GlobalNoSeries.Get(DefaultNoSeriesCode);
+            end;
+            NewNo := GetNextNo(GlobalNoSeries.Code, NewDate, true);
+            NewNoSeriesCode := GlobalNoSeries.Code;
+        end else
+            TestManual(DefaultNoSeriesCode);
+        OnAfterInitSeries(GlobalNoSeries, DefaultNoSeriesCode, NewDate, NewNo);
+    end;
+#endif
```

Comment:
Move this `#endif` below the next two obsoleted functions

Reply:
The functions below are still not completely obsoleted in BaseApp, hence we cannot CLEAN them at this moment, but they will soon.